src/plat/pc99/machine/intel-vtd.c

```diff
@@ -265,6 +293,38 @@ void vtd_handle_fault(void)
     }
 }
 
+void vtd_create_irte(word_t interrupt_index, word_t vector, word_t pci_bus, word_t pci_dev, word_t pci_func)
+{
+    if (interrupt_index >= N_VTD_IRTES_4K) {
+        printf("IOMMU: entry %lu for vector %lu is greater than the size of the IRT, going to ignore it.\n", interrupt_index,
+               vector);
+        return;
+    }
+
+    printf("IOMMU: adding IRTE %lu for vector %lu.\n", interrupt_index, vector);
+
+    /* Remapping hardware units are configured to share the interrupt-remapping table,
+     * so we just read the table address from the first unit
+     */
+    if (x86KSnumDrhu == 0) {
+        return;
+    }
```

Comment:
This check should be done first, otherwise we may get irrelevant and possibly misleading printfs.